### PR TITLE
endpoint for listing namespaces for a cluster

### PR DESCRIPTION
**Pull request by lpuskas**
* The PR contains the implementation of retrieving namespaces for a given cluster (this is required by pipeline clients to facilitate eg.: feature activation)

* the new resource has been added to the open API specification

### CODE REVIEW

File: apis/pipeline/pipeline.yaml [hunk 1]
```diff
@@ -4496,6 +4496,58 @@ paths:
...
+                    required: true
+                    schema:
+                        type: integer
+            responses:
+                '200':
+                    description: Namespaces listed
+                    content:
+                        application/json:
+                            schema:
+                                type: array
```
Comment (sagikazarmark) [inline apis/pipeline/pipeline.yaml]: Returning an array directly is generally not a good idea.

Comment (lpuskas) [reply apis/pipeline/pipeline.yaml]: thx, fixed.

File: cmd/pipeline/main.go [hunk 2]
```diff
@@ -830,7 +830,8 @@ func main() {
 			cRouter.POST("/nodepools/labels", nplsApi.SetNodepoolLabelSets)
 
 			namespaceAPI := namespace.NewAPI(commonClusterGetter, errorHandler)
-			namespaceAPI.RegisterRoutes(cRouter.Group("/namespaces/:namespace"))
+			cRouter.DELETE("/namespaces/:namespace", namespaceAPI.Delete)
```
Comment (siliconbrain) [inline cmd/pipeline/main.go]: Why not use `RegisterRoutes`?

Comment (lpuskas) [reply cmd/pipeline/main.go]: thx,fixed

**Pull request merged**